dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.dotmarketing.factories;
+
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.VersionableAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.business.DotContentletStateException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.util.Logger;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.liferay.portal.model.User;
+
+public class MultiTreeAPIImpl implements MultiTreeAPI {
+
+    final VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+    final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+    final User systemUser = APILocator.systemUser();
+
+
+    public void saveMultiTrees(final String pageId, final List<MultiTree> multiTrees) throws DotDataException {
+        Logger.info(this, String.format("Saving MutiTrees: pageId -> %s multiTrees-> %s", pageId, multiTrees));
+        MultiTreeFactory.saveMultiTrees(pageId, multiTrees);
+    }
+
+    public void saveMultiTree(final MultiTree multiTree) throws DotDataException {
+        Logger.info(this, String.format("Saving MutiTree: %s", multiTree));
+        MultiTreeFactory.saveMultiTree(multiTree);
+    }
+
+    public Table<String, String, Set<String>>  getPageMultiTrees(final IHTMLPage page, final boolean liveMode)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)